graphics/surface.cpp

```diff
@@ -558,7 +558,8 @@ Graphics::Surface *Surface::convertTo(const PixelFormat &dstFormat, const byte *
 
 	// We are here when we are converting from a higher bpp or palettes are different
 	if (dstFormat.bytesPerPixel == 1) {
-		ditherFloyd(srcPalette, srcPaletteCount, surface, dstPalette, dstPaletteCount, method);
+		ditherFloyd(srcPalette, srcPaletteCount, surface, dstPalette, dstPaletteCount, method,
+			    dstFormat == PixelFormat(1, 3, 3, 2, 0, 5, 2, 0, 0));
```

Comment:
Maybe the format should be passed to ditherFloyd and checked for RGB332 inside?

Reply:
Done